Misc/NEWS.d/next/Core and Builtins/2019-02-12-20-16-34.bpo-35961.7f7Sne.rst

```diff
@@ -0,0 +1,2 @@
+Fix a crash in slice_richcompare(): use strong references rather than stolen
```

Comment:
s/stolen/borrowed/

Reply:
Well, @pablogsal and @tim-one said the opposite... I will stay with stolen :-)

Reply:
Oh, I'm sorry about it.  I'm concur with you (https://github.com/python/cpython/pull/11830#discussion_r256125632).
"Owner" has responsibility to "DECREF".  In this case, tuples don't DECREF range members.

But it is not important enough to continue discussion.

Reply:
@methane , please read the docs I already linked to.  After `PyTuple_SET_ITEM()`, it _is_ the tuple's responsibility to decref the item it grabbed.   And it would indeed do so, if it were not for the bizarre-looking code at the end to force all the items to `NULL` without touching their refcounts..  As the docs say, the tuple stole the references from the slice object, and at _that_ point - which is the point at which gc blows up - it was the slice object that was using "borrowed" references.  As the docs say,  `PyTuple_SET_ITEM()` _transfers_ ownership.

Which is the fundamental bug here:  two objects "believe" they own a reference, but the refcount only accounts for one of them.  It's more-or-less arbitrary to pick which one "really" owns it and which "really" borrows it, but the docs are 100% clear about that `PyTuple_SET_ITEM()` made the tuple the owner at the time gc blows up.

Reply:
See https://docs.python.org/3/c-api/intro.html#reference-count-details

"“Owning a reference” means being responsible for calling Py_DECREF on it when the reference is no longer needed."

`PyTuple_SET_ITEM()` is defined as *stole* reference *because* it will be `DECREF`ed when tuple is released.
But in this case, tuple never call DECREF because they will be overwritten soon.

Tuple "believes" it stoled (owns) the reference, but it is "borrowed" reference actually.  That caused gc fail.  GC callback must not be called for borrowed reference.
If tuple really stoled the reference, there were no problem.

Reply:
In other words, "Use strong reference rather than stolen reference" doesn't make sense, because
"stolen reference" is "strong reference".
It was "borrowed reference which tuple believed stolen (strong) reference".